### PR TITLE
Pick the proper fusion gene partner

### PR DESCRIPTION
**Pull request by zhx828**
This is related to https://github.com/cBioPortal/cbioportal/issues/8571

### CODE REVIEW

File: service/src/main/java/org/cbioportal/service/util/MutationMapperUtils.java [hunk 1]
```diff
@@ -113,10 +113,19 @@
...
                                     Gene site2Gene = getGene(site2GeneSymbol);
-                                    if (site2Gene != null) {
-                                        structuralVariant.setSite2EntrezGeneId(site2Gene.getEntrezGeneId());
-                                        structuralVariant.setSite2HugoSymbol(site2Gene.getHugoGeneSymbol());
+                                    // we need to pick the right gene since the fusion gene has been assigned to structural variant site1
+                                    Gene pickedGene = null;
+                                    if (site1Gene != null && !site1Gene.getEntrezGeneId().equals(fusion.getEntrezGeneId())) {
+                                        pickedGene = site1Gene;
+                                    }
+                                    if (site2Gene != null && !site2Gene.getEntrezGeneId().equals(fusion.getEntrezGeneId())) {
```
Comment (kalletlak) [inline service/src/main/java/org/cbioportal/service/util/MutationMapperUtils.java:123]: shouldn't this be in else block or `pickedGene == null` needs to be included in if caluse

Comment (zhx828) [reply service/src/main/java/org/cbioportal/service/util/MutationMapperUtils.java]: @kalletlak updated

File: service/src/main/java/org/cbioportal/service/util/MutationMapperUtils.java [hunk 2]
```diff
@@ -113,10 +113,18 @@
                                     structuralVariant.setSite2EntrezGeneId(fusion.getEntrezGeneId());
                                     structuralVariant.setSite2HugoSymbol(fusion.getGene().getHugoGeneSymbol());
                                 } else {
+                                    Gene site1Gene = getGene(site1GeneSymbol);
                                     Gene site2Gene = getGene(site2GeneSymbol);
-                                    if (site2Gene != null) {
-                                        structuralVariant.setSite2EntrezGeneId(site2Gene.getEntrezGeneId());
-                                        structuralVariant.setSite2HugoSymbol(site2Gene.getHugoGeneSymbol());
+                                    // we need to pick the right gene since the fusion gene has been assigned to structural variant site1
```
Comment (kalletlak) [inline service/src/main/java/org/cbioportal/service/util/MutationMapperUtils.java:118]: `site2Gene` is used if the first condition fails. may be move this inside `else` block

**Review by kalletlak — APPROVED** [review]
LGTM

File: service/src/main/java/org/cbioportal/service/util/MutationMapperUtils.java (hunk 2 above)
Comment (zhx828) [reply service/src/main/java/org/cbioportal/service/util/MutationMapperUtils.java:117]: @kalletlak updated

**Pull request merged**